frontend/document/document.lua

```diff
@@ -207,8 +207,51 @@ function Document:getNativePageDimensions(pageno)
     return page_size
 end
 
-function Document:getProps()
-    return self._document:getDocumentProps()
+function Document:getDocumentProps()
+    -- pdfdocument, djvudocument
+    return self._document:getMetadata()
+    -- credocument, picdocument - overridden by a document implementation
+end
+
+function Document:getProps(props)
```

Comment:
While reading, I find it a bit odd that we could pass a props to this low-level Document method that will actually make it use that and not return the real document props.
I see it is used for documents last opened before 2017 and passing some old `doc_settings:readSetting("stats")` when metadata was saved there.
I guess if we would want to keep that one clean from that, we'd need yet another :getSomethingProps() :) so let's not bother.
Just mentionning that in case you feel the same and would want to think again :)

(Or may be just rename this parameter to something like `use_these_metadata` (a better name than that, but something as explicite)?)

Reply:
Maybe `metadata_to_be_converted`?

Reply:
It is not really "converted" that much :)
May be `cached_metadata` , `cached_doc_metadata` ?